daffodil-runtime1/src/main/scala/org/apache/daffodil/debugger/InteractiveDebugger.scala

```diff
@@ -1567,6 +1635,15 @@ class InteractiveDebugger(runner: InteractiveDebuggerRunner, eCompilers: Express
           }
           DebugState.Pause
         }
+
+        def diff(prestate: StateForDebugger, state: ParseOrUnparseState): Boolean = {
+          if (prestate.groupPos != state.groupPos) {
+            debugPrintln("%s: %d -> %d".format(name, prestate.groupPos, state.groupPos), "  ")
```

Comment:
We are doing this for bassically every diff.

Would it make more sense for the interface to instead be:

```
getValue(state: State): Option[Any]
```
Then let the trait itself implement the diff() function?